lib/rubocop/cop/layout/empty_line_after_guard_clause.rb

```diff
@@ -109,19 +114,23 @@ def next_sibling_empty_or_guard_clause?(node)
         end
 
         def last_argument_is_heredoc?(node)
-          last_children = node.children.last
+          last_children = node.if_branch
 
           return false unless last_children && last_children.send_type?
 
           last_argument = last_argument(node)
 
-          last_argument &&
-            (last_argument.str_type? || last_argument.dstr_type?) &&
-            last_argument.heredoc?
+          heredoc?(last_argument)
         end
 
         def last_argument(node)
-          node.children.last.last_argument
+          node.if_branch.children.last
+        end
+
+        def heredoc?(node)
+          ((node.respond_to?(:str_type?) && node.str_type?) ||
```

Comment:
What's the case where `#last_argument` is not a `Node`? 🤔

Reply:
The method name `#last_argument` may not be representation well.
This code uses `respond_to?(:heredoc?)` because it possible returns `SendNode` or` Symbol` instead of `StrNode`.
I have not yet had a reasonable solution, but I have refactored redundant codes.